Lib/fontbakery/profiles/universal.py

```diff
@@ -287,6 +288,53 @@ def com_google_fonts_check_family_single_directory(fonts):
         )
 
 
+@check(
+    id="com.google.fonts/check/caps_vertically_centered",
+    proposal="https://github.com/googlefonts/fontbakery/issues/4139",
+    rationale="""
+        If possible, the uppercase glyphs should be vertically centered in the em box.
+        This allows for easier centering of text in buttons and grid systems.
+        There is a detailed twitter thread about this subject here:
+        https://twitter.com/romanshamin_en/status/1562801657691672576
+    """,
+)
+def com_google_fonts_check_caps_vertically_centered(ttFont):
+    """Check if uppercase glyphs are vertically centered."""
+
+    filename = os.path.basename(ttFont.reader.file.name)
+
+    # Check both OS/2 and hhea are present.
+    missing_tables = False
+
+    required = ["OS/2", "hhea"]
+    for key in required:
+        if key not in ttFont:
+            missing_tables = True
+            yield FAIL, Message(f"lacks-{key}", f"{filename} lacks a '{key}' table.")
+
+    if missing_tables:
+        return
+
+    upm = ttFont["head"].unitsPerEm
+    error_margin = upm * 0.05
+    cap_height = ttFont["OS/2"].sCapHeight
```

Comment:
The cap height may have been set incorrectly by the designer. This is quite common throughout our collection. A better approach may be to work out the actual height using glyph pens.

Reply:
Good idea, I already figured out how to do this with `fontTools.pens.boundsPen`.

Reply:
I have this as a stand-alone python script doing what I want, but I'm having trouble getting this working in the FontBakery check:
```python
from fontTools.ttLib import TTFont
from fontTools.pens.boundsPen import BoundsPen

f = TTFont(u"Mebo-Regular.ttf")
glyphSet = f.getGlyphSet()

for glyphName in ['A', 'B', 'C', 'D', 'E', 'H', 'I']:
    pen = BoundsPen(glyphSet)
    glyphSet[glyphName].draw(pen)
    print(glyphName, pen.bounds)
    highest_point = pen.bounds[3]
    print(f"The highest point in the specified glyphs is: {highest_point}")
```

Reply:
This is now fixed in commit [b9aad04](https://github.com/googlefonts/fontbakery/pull/4226/commits/b9aad04a90c3b6766810b3baec6303093c8db2f7)